pkg/kncloudevents/utils.go

```diff
@@ -26,6 +26,34 @@ import (
 	"github.com/cloudevents/sdk-go/v2/types"
 )
 
+const (
+	// AcceptReplyHeaderKey is the key for the event reply header.
+	// https://github.com/knative/eventing/blob/master/docs/spec/data-plane.md#event-reply-contract
+	AcceptReplyHeaderKey = "Prefer"
+	// AcceptReply is the value to accept event reply.
+	AcceptReply = "reply"
+)
+
+var acceptReplyHeaders = []string{AcceptReply}
+
+// SetAcceptReplyHeader sets Prefer: reply.
+func SetAcceptReplyHeader(headers nethttp.Header) {
```

Comment:
no need to do any of this, just use `headers.Set`

Reply:
`Headers.Set` was the originally used, but wouldn't we want to keep existing ones?

Reply:
Set appends to the array if they key is not present. add a test to confirm

Reply:
Set and Add will simply call the following. I have considered them both, `Set` will overwrite all the existing ones, which will work in our case now as we are initiating the event, but if later we need more Prefer headers we need to be very careful here.. `Add` will not insert new key from the code.
```
func (h MIMEHeader) Add(key, value string) {
    key = CanonicalMIMEHeaderKey(key)
    h[key] = append(h[key], value)
}

// Set sets the header entries associated with key to
// the single element value. It replaces any existing
// values associated with key.
func (h MIMEHeader) Set(key, value string) {
    h[CanonicalMIMEHeaderKey(key)] = []string{value}
}
```